cpg_pipes/providers/inputs.py

```diff
@@ -0,0 +1,417 @@
+    def get_dataset_name(self, entry: dict) -> str | None:
+        """
+        Get name of the dataset.
+        """
+        return entry.get(FieldMap.dataset.value)
+
+    # noinspection PyMethodMayBeStatic
+    def get_sample_id(self, entry: dict[str, str]) -> str:
+        """
+        Get sample ID from a sample dict.
+        """
+        if 'sample' not in entry:
+            raise InputProviderError(
+                f'Metadata TSV must have a header and a column '
+                f'"{FieldMap.sample.value}", got: {entry}'
+            )
+        return entry[FieldMap.sample.value]
+
+    # noinspection PyMethodMayBeStatic
+    def get_external_id(self, entry: dict) -> str | None:
+        """
+        Get external sample ID from a sample dict.
+        """
+        return entry.get(FieldMap.external_id.value, None)
+
+    # noinspection PyMethodMayBeStatic
+    def get_participant_id(self, entry: dict) -> str | None:
+        """
+        Get participant ID from a sample dict.
+        """
+        return entry.get(FieldMap.participant_id.value, None)
+
+    # noinspection PyMethodMayBeStatic
+    def get_participant_sex(self, entry: dict) -> Sex:
+        """
+        Get participant's sex/gender from a sample dict.
+        """
+        return Sex.parse(entry.get(FieldMap.sex.value, None))
+
+    # noinspection PyMethodMayBeStatic
+    def get_sample_meta(self, entry: dict) -> dict:
+        """
+        Get sample metadata.
+        """
+        reserverd_fields = [f.value for f in FieldMap]
+        return {k: v for k, v in entry.items() if k not in reserverd_fields}
+
+    def get_sequencing_type(self, entry: dict) -> SequencingType:
+        """
+        Get sequencing type.
+        """
+        return SequencingType.parse(entry[FieldMap.sequencing_type.value])
+
+    def populate_analysis(self, cohort: Cohort) -> None:
+        """
+        Populate Analysis entries.
+        """
+        pass
+
+    def populate_participants(self, cohort: Cohort) -> None:
+        """
+        Populate Participant entries.
+        """
+        pass
+
+    def populate_pedigree(self, cohort: Cohort) -> None:
+        """
+        Populate pedigree data.
+        """
+        pass
+
+    def populate_alignment_inputs(self, cohort: Cohort) -> None:
+        """
+        Populate sequencing inputs for samples.
+        """
+        d_by_sid: dict[str, AlignmentInput] = {}
+
+        for entry in self.get_entries():
+            sid = self.get_sample_id(entry)
+            fqs1 = [
+                f.strip()
+                for f in entry.get(FieldMap.fqs_r1.value, '').split('|')
+                if f.strip()
+            ]
+            fqs2 = [
+                f.strip()
+                for f in entry.get(FieldMap.fqs_r2.value, '').split('|')
+                if f.strip()
+            ]
+            cram = entry.get(FieldMap.cram.value, None)
+            if fqs1:
+                if len(fqs1) != len(fqs1):
```

Comment:
I think mypy should catch this:
```suggestion
                if len(fqs1) != len(fqs2):
```

Reply:
`mypy` is just a type checker, it wouldn't catch that unfortunately :( Thanks!